### ISSUE TITLE
undefined method `<' for nil:NilClass

### ISSUE THREAD

**Issue by kations** [locomotivecms/engine#1179]
Hi,

i have pulled a Site from V2 and successfully pusht to V3. Than i pulled the Site from V3 make some changes and try to push it again. Now this Error appears and i can't find anything related in the Template.

Pushing Pages
  persisting index (de) [failed]
    Locomotive Back-office error. Contact your administrator or check your application logs.
# Error description:

undefined method `<' for nil:NilClass
# Backtrace:

```
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_coal-1.2.0/lib/locomotive/coal/resources/concerns/request.rb:36:in `do_request'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_coal-1.2.0/lib/locomotive/coal/resources/concerns/request.rb:7:in `get'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_coal-1.2.0/lib/locomotive/coal/resources/base.rb:17:in `index'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_assets_command.rb:46:in `block in remote_entities'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_assets_command.rb:45:in `tap'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_assets_command.rb:45:in `remote_entities'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_assets_command.rb:39:in `remote_entity'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_content_assets_command.rb:18:in `persist'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/persist_assets_concern.rb:11:in `block in replace_with_content_assets!'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/persist_assets_concern.rb:10:in `gsub'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/persist_assets_concern.rb:10:in `replace_with_content_assets!'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/page_decorator.rb:82:in `template'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/to_hash_concern.rb:10:in `public_send'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/to_hash_concern.rb:10:in `block (2 levels) in to_hash'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/to_hash_concern.rb:9:in `each'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/to_hash_concern.rb:9:in `block in to_hash'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/to_hash_concern.rb:8:in `tap'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/decorators/concerns/to_hash_concern.rb:8:in `to_hash'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:27:in `block in persist'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:111:in `block (3 levels) in translated_in'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:110:in `tap'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:110:in `block (2 levels) in translated_in'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_steam-1.1.1/lib/locomotive/steam/decorators/i18n_decorator.rb:48:in `__with_locale__'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:109:in `block in translated_in'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:108:in `each'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:108:in `find_all'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:108:in `translated_in'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_pages_command.rb:23:in `persist'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:34:in `block in _push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:30:in `each'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:30:in `_push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:25:in `block in _push_with_timezone'
/Library/Ruby/Gems/2.0.0/gems/activesupport-4.2.6/lib/active_support/core_ext/time/zones.rb:48:in `use_zone'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:24:in `_push_with_timezone'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:18:in `block in push'
/Library/Ruby/Gems/2.0.0/gems/activesupport-4.2.6/lib/active_support/notifications.rb:166:in `instrument'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:50:in `instrument'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:16:in `push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_sub_commands/push_base_command.rb:12:in `push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:55:in `block in _push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:84:in `block in each_resource'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:77:in `each'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:77:in `each_resource'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:54:in `_push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:41:in `push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/commands/push_command.rb:31:in `push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon.rb:73:in `push'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/lib/locomotive/wagon/cli.rb:318:in `deploy'
/Library/Ruby/Gems/2.0.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
/Library/Ruby/Gems/2.0.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
/Library/Ruby/Gems/2.0.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
/Library/Ruby/Gems/2.0.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
/Library/Ruby/Gems/2.0.0/gems/locomotivecms_wagon-2.1.1/bin/wagon:12:in `<top (required)>'
/usr/local/bin/wagon:23:in `load'
/usr/local/bin/wagon:23:in `<top (required)>'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `load'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:63:in `kernel_load'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/cli/exec.rb:24:in `run'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/cli.rb:304:in `exec'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/cli.rb:11:in `start'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/exe/bundle:27:in `block in <top (required)>'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/lib/bundler/friendly_errors.rb:98:in `with_friendly_errors'
/Library/Ruby/Gems/2.0.0/gems/bundler-1.12.5/exe/bundle:19:in `<top (required)>'
/usr/local/bin/bundle:23:in `load'
/usr/local/bin/bundle:23:in `<main>'
```

**Comment by did** [comment]
hi @kations, what did your server respond to the last API request (to /api/content_assets)?

**Comment by kations** [comment]
@did 
I get the error also on locomotive.works, where i find the error? Developer/Errors shows no error.
Locomotive.works Site: https://station.locomotive.works/_app/zealous-dusk-91/preview

Local the Site runs without errors.

**Comment by kations** [comment]
@did 
Seems the Error is related to {% editable_file "Bild 4" %}/samples/assets/IMG_5622-1.jpg{% endeditable_file %}

If i delete the direct path in the editable File the push works again.

A related question, how i can delete a Snipped from a Site in V3? Or a unused File? If i delete Stuff local and than push/pull the File still exists.

**Comment by did** [comment]
Hmmm, I can't find your exception in our server logs. Could you try again please?

Regarding your last question, you'll find answers here: https://locomotive-v3.readme.io/docs/delete-resources.

**Comment by kations** [comment]
Also get this Error now by pulling a Site, do you have an idea how i can resolve this?

Pulling ContentAssets

# Error description:
undefined method `<' for nil:NilClass

I think it would be very useful for Admins to Edit Files in the Backoffice, maybe in Settings a Tab with Files where Admins find a File viewer. Something like this, where you can see just the file structure and be able to delete and edit some files:

http://docs.webix.com/samples/64_file_manager/01_basic/03_uploading.html#!/i18n

What do you think?

**Comment by conchulio** [comment]
Hey, I ran into exactly the same error. It's very mysterious to me what causes it but applying the patch I posted at least solved it for me: https://github.com/locomotivecms/coal/pull/21

**Comment by cappadona** [comment]
I saw this `undefined method '<' for nil:NilClass` on a recent local install of the Engine (`v3.3.0.rc2`) as well when deploying via wagon.

Tailing `log/development.log` showed the same error, but tailing `log/steam.log` revealed an underlying issue with Dragonfly:

```sh
DRAGONFLY: caught error - Command failed ('/usr/local/bin/convert' '<pathToFile>' '-resize' '70x70^^' '-gravity' 'Center' '-crop' '70x70+0+0' '+repage' '/var/folders/_4/jj7dg2450ns3y_dj1hc0z8wc0000gn/T/dragonfly20170330-43510-18wno6w.jpg') with exit status 127 and stderr sh: /usr/local/bin/convert: No such file or directory
```

The solution was to install `imagemagick`, which is [a requirement](http://markevans.github.io/dragonfly/imagemagick) due to [Locomotive's dragonfly config](https://github.com/locomotivecms/engine/blob/master/lib/generators/locomotive/install/templates/dragonfly.rb#L3).

I'd be surprised if this is the same underlying issue when dealing with `locomotive.works`, but for anyone running the engine themselves, I figured it was worth sharing since @kations issue does appear to be image related.
